lib/ui/views/product_view.dart

```diff
@@ -0,0 +1,93 @@
+      viewModelBuilder: () => ProductViewModel(),
+      onModelReady: (model) {
+        model.product = product;
+      },
+      builder: (context, model, child) => Scaffold(
+        appBar: AppBar(
+          title: const Text("Product"),
+        ),
+        body: model.busy
+            ? const BusyIndicator()
+            : Column(
+                mainAxisAlignment: MainAxisAlignment.start,
+                crossAxisAlignment: CrossAxisAlignment.stretch,
+                children: [
+                  Image.network(product.imageUrl ?? Product.missingPhotoUrl,
+                      fit: BoxFit.cover, height: 150),
+                  Padding(
+                      padding: const EdgeInsets.fromLTRB(15, 5, 15, 0),
+                      child: Column(
+                          mainAxisAlignment: MainAxisAlignment.start,
+                          crossAxisAlignment: CrossAxisAlignment.start,
+                          children: [
+                            Text(product.name, style: titleStyleHugeB),
+                            const Text('Allergens:')
+                          ])),
+                  product.allergens.isNotEmpty
+                      ? Expanded(
+                          child: ListView.builder(
+                          itemExtent: 60.0,
+                          itemBuilder: (context, index) =>
+                              buildAllergenList(context, index, model, product),
+                          itemCount: product.allergens.length,
+                        ))
+                      : const Text("Haven't found any allergens"),
+                  TextButton(
+                      style: const ButtonStyle(),
+                      child: const Text(
+                        'Placeholder button.',
+                        style: TextStyle(
+                            color: primaryColor, fontWeight: FontWeight.bold),
+                      ),
+                      onPressed: () => {}),
+                  TextButton(
+                      child: const Text(
+                        'Placeholder button2.',
+                        style: TextStyle(
+                            color: secondaryColor,
+                            fontWeight: FontWeight.normal),
+                      ),
+                      onPressed: () => {})
+                ],
+              ),
+      ),
+    );
+  }
+
+  Widget buildAllergenList(
+      BuildContext context, int index, ProductViewModel model, Product p) {
+    return Padding(
+        padding: const EdgeInsets.symmetric(horizontal: 20),
+        child: Row(
+          mainAxisAlignment: MainAxisAlignment.spaceBetween,
+          children: [
+            Row(
+              mainAxisAlignment: MainAxisAlignment.spaceBetween,
+              children: [Text(p.allergens[index]), horizontalSpaceMedium],
+            ),
+            ElevatedButton(
+              onPressed: () => {},
+              child: const Text('placeholder'),
+            ),
+          ],
+        ));
```

Comment:
jeśli text będzie za długi to nie zmieści się na ekranie 
ten wewnętrzy row nie jest raczej potrzebny
child: Row(
        mainAxisAlignment: MainAxisAlignment.spaceBetween,
        children: [
          Expanded(child: Text(p.allergens[index])),
          ElevatedButton(
            onPressed: () => {},
            child: const Text('placeholder'),
          ),
        ],
      ),
takie coś powinno załatwić tą samą robotę, a nie będzie powodować overflow w row, bo jeśli wiersz będzie za długi to text się zawinie do następnej linii

Reply:
nice, dzięki
fixed